frontend/public/components/pod.tsx

```diff
@@ -551,7 +552,60 @@ const PodGraphs = requirePrometheus(({ pod }) => {
   );
 });
 
-export const PodStatus: React.FC<PodStatusProps> = ({ pod }) => <Status status={podPhase(pod)} />;
+const PodStatusPopover: React.FC<PodStatusPopoverProps> = ({
+  bodyContent,
+  headerContent,
+  status,
+}) => {
+  return (
+    <Popover headerContent={headerContent} bodyContent={bodyContent}>
+      <Button variant="link" isInline>
+        <Status status={status} />
+      </Button>
+    </Popover>
+  );
+};
+
+export const PodStatus: React.FC<PodStatusProps> = ({ pod }) => {
+  const status = podPhase(pod);
+  const unschedulableCondition = pod.status?.conditions?.find(
+    (condition) => condition.reason === 'Unschedulable' && condition.status === 'False',
+  );
+  const containerStatusImagePullBackOff = pod.status?.containerStatuses?.find(
+    (cs) => cs.state?.waiting?.reason === 'ImagePullBackOff',
+  );
+  const containerStatusErrImagePull = pod.status?.containerStatuses?.find(
+    (cs) => cs.state?.waiting?.reason === 'ErrImagePull',
+  );
```

Comment:
Do you see any weird behavior if you remove the `reason` check and just always show `state.waiting.message` if present?

Reply:
Good call.  I was able to do that and DRY up the code significantly.